### PR TITLE
Accelerator: Add Spoke deployment to hubAndSpoke pipeline

### PR DESCRIPTION
**Pull request by baartch**
# Overview/Summary
Regarding the Hub-and-spoke topology, by now the Accelerator is not able to deploy any spoke networks.
I need this functionality, that's why I added another pipeline step.
Probably you would have solved this with a different approach. So I'm courious about the teams comments.

## This PR fixes/adds/changes/removes
I added an additional step to the pipeline _4a-hubandspoke_.
This step calls the [_hubPeeredSpoke_ Orchestration Module](https://github.com/Azure/ALZ-Bicep/tree/main/infra-as-code/bicep/orchestration/hubPeeredSpoke)
This step takes every parameter file matching the following:  
`config/custom-parameters/hubPeeredSpoke.*.parameters.all.json`

This allows me to add as many parameter files as I need.
- `config/custom-parameters/hubPeeredSpoke.MYSPOKE1.parameters.all.json`
- `config/custom-parameters/hubPeeredSpoke.MYSPOKE2.parameters.all.json`
- and so on...

### Breaking Changes

None

## Testing Evidence

image

## As part of this Pull Request I have

- [x] Read the [Contribution Guide](https://github.com/Azure/ALZ-Bicep/wiki/Contributing) and ensured this PR is compliant with the guide
- [x] Ensured the resource API versions in `.bicep` file/s I am adding/editing are using the latest API version possible
- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/ALZ-Bicep/pulls)
- [ ] Associated it with relevant [GitHub Issues](https://github.com/Azure/ALZ-Bicep/issues)
- [ ] *(ALZ Bicep Core Team Only)* Associated it with relevant [ADO Items](https://aka.ms/alz/bicep/backlog)
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/ALZ-Bicep/tree/main)
- [x] Performed testing and provided evidence.
- [ ] Updated one or more of the following tests *(if required)*
  - [Unit](https://github.com/Azure/ALZ-Bicep/blob/main/.github/workflows/bicep-build-to-validate.yml)
  - [Linting](https://github.com/Azure/ALZ-Bicep/tree/main/.github/workflows)
  - [E2E (End-To-End)](https://github.com/Azure/ALZ-Bicep/blob/main/tests/pipelines/bicep-build-to-validate.yml)
  - [ValidateAzCloud (Base validation in Azure Cloud)](https://github.com/Azure/ALZ-Bicep/blob/main/tests/pipelines/base-unit-validate.yml)
  - [ValidateMcCloud (Base validation in Azure China Cloud)](https://github.com/Azure/ALZ-Bicep/blob/main/tests/pipelines/mc-base-unit-validate.yml)
- [ ] Updated relevant and associated documentation (e.g. Contribution Guide, Module READMEs, Wiki Docs etc.)
- [ ] If relevant, created or updated Code Tours [here](https://github.com/Azure/ALZ-Bicep/blob/main/.vscode/tours)

### CODE REVIEW

File: infra-as-code/bicep/orchestration/hubPeeredSpoke/hubPeeredSpoke.bicep [hunk 1]
```diff
@@ -17,7 +17,7 @@ param parTopLevelManagementGroupPrefix string = 'alz'
 @maxLength(10)
 param parTopLevelManagementGroupSuffix string = ''
 
-@sys.description('Subscription Id to the Virtual Network Hub object. Default: Empty String')
```
Comment (baartch) [inline infra-as-code/bicep/orchestration/hubPeeredSpoke/hubPeeredSpoke.bicep:20]: It confused me alot that this is called Hub. It is the subscriptionId where the spoke is placed.
Or am I wrong?

Comment (oZakari) [reply infra-as-code/bicep/orchestration/hubPeeredSpoke/hubPeeredSpoke.bicep:20]: @baartch you are correct, the description should be changed as you mentioned in this PR. Good catch!

**Comment by oZakari** [comment]
Hi @baartch, thanks for bringing this up! We decided not to include this initially as the spoke connectivity has been delegated to the [bicep-lz-vending](https://github.com/Azure/bicep-lz-vending) module which is oriented towards the application landing zones.

There is an item in our backlog to integrate the Accelerator workflows/pipeleins with the bicep-lz-vending module, but we're trying to get some core features sorted out with the accelerator in relation to this repo before then.

As a result, we'll hold off on this for now, but feel free to open another PR with the param description fix otherwise I can get it included in a PR tomorrow as well.

**Comment by baartch** [comment]
Ok. Thank you. I didn't know about the _bicep-lz-vending_ repository. Unfortunately there is no mention of it in the _ALZ-Bicep_ documentation.

**Pull request closed without merging**